sdk/core/azure-core/inc/http/curl/curl.hpp

```diff
@@ -475,11 +472,11 @@ namespace Azure { namespace Core { namespace Http {
      * @param count number of bytes to copy from network into buffer.
      * @return the number of read and copied bytes from network to buffer.
      */
-    uint64_t Read(uint8_t* buffer, uint64_t count) override
+    int64_t Read(uint8_t* buffer, int64_t count) override
     {
-      if (this->m_length == this->m_offset && !this->m_unknownSize)
+      if (this->m_length < 0)
```

Comment:
This doesn't look right. If the length is <0, then you'll never read anything, right?

Reply:
length == -1 means unknown size. So Read is performed as many times until we found the end of body delimiter or the connection is closed. 
This is only an implementation on BodyStream from transporters like libcurl